feature_layers/display-clusters/src/main/java/com/esri/samples/display_clusters/DisplayClustersSample.java

```diff
@@ -0,0 +1,171 @@
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.popup.Popup;
+import com.esri.arcgisruntime.mapping.view.Callout;
+import com.esri.arcgisruntime.mapping.view.IdentifyLayerResult;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import javafx.application.Application;
+import javafx.geometry.Point2D;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.layout.VBox;
+import javafx.scene.web.WebEngine;
+import javafx.scene.web.WebView;
+import javafx.scene.control.Button;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+
+public class DisplayClustersSample extends Application {
+
+  private ArcGISMap map;
+  private MapView mapView;
+  private PortalItem portalItem; // keep loadable in scope to avoid garbage collection
+  private FeatureLayer powerPlantsLayer;
+
+  @Override
+  public void start(Stage stage) {
+
+    try {
+      // create stack pane and application scene
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Display Points Using Clustering Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
+      // authentication with an API key or named user is required to access basemaps and other location services
+      String yourAPIKey = System.getProperty("apiKey");
+      ArcGISRuntimeEnvironment.setApiKey(yourAPIKey);
+
+      // create a button to toggle clustering
+      Button featureClusteringToggle = new Button("Toggle Feature Clustering");
+      featureClusteringToggle.setDisable(true);
+
+      // create a portal and portal item, using the portal and item ID
+      var portal = new Portal("https://www.arcgis.com/");
+      portalItem = new PortalItem(portal, "8916d50c44c746c1aafae001552bad23");
+
+      map = new ArcGISMap(portalItem);
+      map.addDoneLoadingListener(() -> {
+        if (map.getLoadStatus() == LoadStatus.LOADED) {
+          featureClusteringToggle.setDisable(false);
+
+          powerPlantsLayer = (FeatureLayer)map.getOperationalLayers().get(0);
+
+          // when the button is clicked, toggle the feature clustering
+          mapView.setOnMouseClicked(mouseEvent -> {
+            if (mapView == null) {
+              return;
+            }
+
+            mapView.getCallout().setVisible(false);
+            Point2D point = new Point2D(mouseEvent.getX(), mouseEvent.getY());
+            ListenableFuture<IdentifyLayerResult> identifiedLayerResults = mapView.identifyLayerAsync(powerPlantsLayer, point, 3.0, false);
+            identifiedLayerResults.addDoneListener(() -> {
+              try {
+                if (identifiedLayerResults.get() == null || identifiedLayerResults.get().getPopups().isEmpty()) {
```

Comment:
I'd consider calling `identifiedLayerResults.get() ` just once and assigning to a local variable.

Also, perhaps best practice to encourage folk to use the `get` which takes a timeout:

```java
var results = identifiedLayerResults.get(10, TimeUnit.SECONDS);
```